ServiceMonitoringFunction/tst/com.amazon.vas.serviceordermonitoringlambda/component/GetJobMetricsComponentTest.java

```diff
@@ -0,0 +1,61 @@
+package component;
+
+import com.amazon.vas.serviceordermonitoringlambda.builder.JobDetailsBuilder;
+import com.amazon.vas.serviceordermonitoringlambda.component.GetJobMetricsComponent;
+import com.amazon.vas.serviceordermonitoringlambda.model.GetJobMetricsInputBO;
+import com.amazon.vas.serviceordermonitoringlambda.model.GetJobMetricsOutputBO;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import util.DefaultModelBuilders;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static util.TestConstants.CITY;
+import static util.TestConstants.MERCHANT_ID;
+
+public class GetJobMetricsComponentTest {
+
+    private DefaultModelBuilders defaultModelBuilders;
+
+    @Mock
+    private JobDetailsBuilder jobDetailsBuilder;
+
+    @InjectMocks
+    private GetJobMetricsComponent getJobMetricsComponent;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testGetJobMetricsComponent_withCityAndSlotStartTime() {
```

Comment:
here our objective is to test 2 things. 
1) Filtering working as expected.
2) Grouping working as expected.

You, need to add testcases where you test both for at least 1-1 keys. I can see that defaultModelBuilders.buildGetJobMetricsInputBO just include grouping criteria but not have filtering key. So, instead of using defaultModelBuilders.buildGetJobMetricsInputBO in tests, you can create appropriate GetJobMetricsInputBO locally inside tests only and do assertion.